main.go

```diff
@@ -365,21 +90,49 @@ func main() {
 
 	m, err := firecracker.NewMachine(vmmCtx, fcCfg, machineOpts...)
 	if err != nil {
-		log.Fatalf("Failed creating machine: %s", err)
+		return fmt.Errorf("Failed creating machine: %s", err)
 	}
 
-	if metadata != nil {
-		m.EnableMetadata(metadata)
+	if opts.validMetadata != nil {
+		m.EnableMetadata(opts.validMetadata)
 	}
 
 	if err := m.Start(vmmCtx); err != nil {
-		log.Fatalf("Failed to start machine: %v", err)
+		return fmt.Errorf("Failed to start machine: %v", err)
 	}
 	defer m.StopVMM()
 
 	// wait for the VMM to exit
 	if err := m.Wait(vmmCtx); err != nil {
-		log.Fatalf("Wait returned an error %s", err)
+		return fmt.Errorf("Wait returned an error %s", err)
 	}
 	log.Printf("Start machine was happy")
+	return nil
+}
+
+func main() {
+	opts := options{}
+	p := flags.NewParser(&opts, flags.Default)
+	// if no args just print help
+	if len(os.Args) == 1 {
+		p.WriteHelp(os.Stderr)
+		os.Exit(0)
+	}
+	_, err := p.ParseArgs(os.Args)
+	if err != nil {
+		// ErrHelp indicates that the help message was printed so we
+		// can exit
+		if val, ok := err.(*flags.Error); ok && val.Type == flags.ErrHelp {
+			os.Exit(0)
+		} else {
+			p.WriteHelp(os.Stderr)
+			os.Exit(1)
```

Comment:
Do we want to print error details to give user some context what exactly was wrong?